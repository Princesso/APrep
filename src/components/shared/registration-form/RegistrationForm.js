import React from 'react'
import '../../../styles/main.css'
import './registration-form.css'
import {Link} from 'react-router-dom'

const RegistrationForm = () => {
  return( 
    <div className="form-holder">
      <form className="form"> 
        <h3>Register</h3>
        <input type="email" placeholder="Email" required id="email"></input>
        <input type="text" placeholder="Username" required id="username"></input>
        <input type="text" placeholder="Firstname" required id="firstname"></input>
        <input type="text" placeholder="Lastname" required id="lastname"></input>
        <input type="password" placeholder="Password" required id="password"></input>
        <button type="submit" id="submitBtn">Login</button>
        <p id="message">Already have an account? <Link to="/login" className="link"> Login</Link></p>
      </form>
    </div>
  )
}

export default RegistrationForm