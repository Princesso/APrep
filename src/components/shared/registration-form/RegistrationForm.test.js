import React from 'react';
import { RegistrationForm } from './RegistrationForm';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import Store from '../../../store';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { mount, shallow } from 'enzyme';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

Enzyme.configure({ adapter: new Adapter() });

test('Registration Form', () => {
  const component = renderer.create(
    <Provider store={Store}>
      <BrowserRouter>
          <Switch>
            <RegistrationForm/>
          </Switch>
      </BrowserRouter>
    </Provider>
  )

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('it should handle submit when register button is clicked', ()=> {
  const preventDefault = jest.fn();

  const wrapper = shallow(<RegistrationForm/>)
  wrapper.find('form').simulate('submit', { preventDefault });
  expect(preventDefault).toHaveBeenCalled();
});

test('it should handle change when input is added', ()=> {
  const target = { 
    name: 'username',
    value: 'ada'
  }

  const wrapper = shallow(<RegistrationForm/>)
  wrapper.find('input').at(0).simulate('change', { target });
  expect(wrapper.state(target.name)).toEqual(target.value);
});