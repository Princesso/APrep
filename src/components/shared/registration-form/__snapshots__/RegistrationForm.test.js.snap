// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Form 1`] = `
<div
  className="form-holder"
>
  <form
    className="form"
    method="post"
    onSubmit={[Function]}
  >
    <div>
      <p />
    </div>
    <h3>
      Register
    </h3>
    <input
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
      value=""
    />
    <input
      id="username"
      name="username"
      onChange={[Function]}
      placeholder="Username"
      required={true}
      type="text"
      value=""
    />
    <input
      id="firstname"
      name="firstname"
      onChange={[Function]}
      placeholder="Firstname"
      required={true}
      type="text"
      value=""
    />
    <input
      id="lastname"
      name="lastname"
      onChange={[Function]}
      placeholder="Lastname"
      required={true}
      type="text"
      value=""
    />
    <input
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <button
      className="submitBtn"
      id="submitBtn"
      type="submit"
    >
      <span
        className="loading"
      >
         
        <p>
          Register
        </p>
      </span>
    </button>
    <p
      className="message"
    >
      Already have an account? 
      <a
        className="linker"
        href="/login"
        onClick={[Function]}
      >
         Login
      </a>
    </p>
  </form>
</div>
`;
